apps/student-app/src/app/student-app.tsx

```diff
@@ -0,0 +1,379 @@
+            <b>{word_list.name}:</b> Words: {word_list.words.length}
+            </button>
+        </li>
+      );
+    });
+
+  }
+
+  // Creates a list of words that were in the game.
+  // Populates a UL element on game-words page.
+  function get_words_from_game() {
+    return words_in_game.map((word: Word) => {
+      return (
+        <li id={`word-${word.word}`}  key={`word-${word.word}`}>
+          <b className={words_correct.includes(word) ? 'correct-word' : 'incorrect-word'}>{word.word}:</b> {word.hint}
+        </li>
+      );
+    });
+  }
+
+  // Button to go to Crouse page
+  function handleToCourse(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.course_state);
+    } catch (err) {
+      setError('error-word-list');
+    }
+  }
+
+  // Back button to return to Splash page
+  function handleBackToSplash(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.splash_state);
+    } catch (err) {
+      setError('error-course');
+    }
+  }
+
+  // Back button to return to Course Code page
+  function handleBackToCourse(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.course_state);
+    } catch (err) {
+      setError('error-word-list');
+    }
+  }
+
+  // Back button to return to Word List page
+  function handleBackToWordLists(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.word_list_state);
+    } catch (err) {
+      setError('error-word-list');
+    }
+  }
+
+  // Back button to return to Score page
+  function handleBackToScore(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.score_state);
+    } catch (err) {}
+  }
+
+  // Replay button that resets the game and starts it again.
+  function handleResetGame(e: any): void {
+    try {
+      setError('null');
+      start_new_game();
+      setScreen(StudentState.game_state);
+    } catch (err) {}
+  }
+
+  // Button to visit Word Review page
+  function handleToGameWords(e: any): void {
+    try {
+      setError('null');
+      setScreen(StudentState.review_words_state);
+    } catch (err) {}
+  }
+
+  // Handles course code submission
+  async function handleSubmitCourse(e: React.FormEvent<HTMLFormElement>): Promise<void> {
+    e.preventDefault();
+    try {
+      // GET on course GUID
+      // STUB getting the data
+      setData(await submitCourse(course_code));
+      populate_word_list_items(); //Create the word lists List Items elements
+      setError('null');
+      setScreen(StudentState.word_list_state);
+    } catch (err) {
+      setError('error-course');
+    }
+  }
+
+  // Handles setting the word list to chosen list item ID
+  async function handleSetWordList(
+    id: string
+  ): Promise<void> {
+    try {
+      ;
+      setSelectedWordList(data?.word_lists.find((wrd_lst: WordList) => {
+        return wrd_lst.id === id;
+      }) ?? null)
+      setError('null');
```

Comment:
i am guessing here is where you will change the state when you are ready to the game starting state

Reply:
Yup, I call start_new_game() and put the change screen state in there,